kanalony-storage-logic/src/main/scala/kanalony/storage/logic/QueryLocator.scala

```diff
@@ -0,0 +1,45 @@
+package kanalony.storage.logic
+
+import kanalony.storage.logic.queries.model.QueryConstraint
+
+/**
+ * Created by elad.benedict on 2/16/2016.
+ */
+object QueryLocator {
+
+  def tableAndQueryEqualityConstraintsMatch(tq: IQuery, queryParams: QueryParams) : Boolean = {
+    val tableEqualityConstrainedDefs = tq.dimensionInformation.filter(_.constraint.constraint == QueryConstraint.Equality).map(_.dimension).toSet
+    val queryEqualityConstraintDefs = queryParams.dimensionDefinitions.filter(_.constraint.constraint == QueryConstraint.Equality).map(_.dimension)
+    queryEqualityConstraintDefs == tableEqualityConstrainedDefs
+  }
+
+  def tableSupportsAllQueryDimensions(tq: IQuery, queryParams: QueryParams) = {
+    val queryDimnesions = queryParams.dimensionDefinitions.map(_.dimension)
+    val tableDimensions = tq.dimensionInformation.map(_.dimension).toSet
+    queryDimnesions subsetOf tableDimensions
+  }
+
+  def calcTableCompatibilityDistance(table: IQuery, queryParams : QueryParams) : Int = {
+    var distance = Int.MaxValue
+    if (!(table.supportedMetrics contains queryParams.metric)){
+      return distance
+    }
+    if (!(tableAndQueryEqualityConstraintsMatch(table, queryParams)))
+    {
+      return distance
+    }
+
+    if (!(tableSupportsAllQueryDimensions(table, queryParams)))
+    {
+      return distance
+    }
```

Comment:
Did you intentionally create 3 If statements instead of a single If with 3 conditions? Maybe I'm missing something but here is how I would write it (sorry about indentation, couldn't make it work :( )
`if (!(table.supportedMetrics contains queryParams.metric) ||`
`!(tableAndQueryEqualityConstraintsMatch(table, queryParams)) ||`
`!(tableSupportsAllQueryDimensions(table, queryParams))) {`
`    Int.MaxValue`
`}`
`else {`
`table.dimensionInformation.length - queryParams.dimensionDefinitions.size`
`}`